docs/mvc/tutorials/mvc-with-entity-framework/sample/src/ContosoBooks/Controllers/BookController.cs

```diff
@@ -0,0 +1,159 @@
+            {
+                if (ModelState.IsValid)
+                {
+                    DbContext.Books.Add(book);
+                    await DbContext.SaveChangesAsync();
+                    return RedirectToAction("Index");
+                }
+            }
+            catch (DataStoreException ex)
+            {
+                Logger.LogError("Unable to save changes.", ex);
+                ModelState.AddModelError(string.Empty, "Unable to save changes.");
+            }
+            return View(book);
+        }
+
+        [HttpGet]
+        [ActionName("Delete")]
+        public async Task<ActionResult> ConfirmDelete(int id, bool? retry)
+        {
+            Book book = await FindBookAsync(id);
+            if (book == null)
+            {
+                return HttpNotFound();
+            }
+            ViewBag.Retry = retry ?? false;
+            return View(book);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> Delete(int id)
+        {
+            try
+            {
+                Book book = await FindBookAsync(id);
+                DbContext.Books.Remove(book);
+                await DbContext.SaveChangesAsync();
+            }
+            catch (DataStoreException ex)
+            {
+                Logger.LogError("Unable to delete record.", ex);
+                return RedirectToAction("Delete", new { id = id, retry = true });
+            }
+            return RedirectToAction("Index");
+        }
+
+        [HttpGet]
+        public async Task<ActionResult> Edit(int id)
+        {
+            Book book = await FindBookAsync(id);
+            if (book == null)
+            {
+                return HttpNotFound();
+            }
+
+            ViewBag.Items = GetAuthorsListItems(book.AuthorID);
+            return View(book);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> Update(int id, [Bind("Title", "Year", "Price", "Genre", "AuthorID")] Book book)
+        {
+            try
+            {
+                book.BookID = id;
+                DbContext.Books.Attach(book);
+                DbContext.Entry(book).State = EntityState.Modified;
+                await DbContext.SaveChangesAsync();
+                return RedirectToAction("Index");
+            }
+            catch (DataStoreException ex)
+            {
+                Logger.LogError("Unable to update.", ex);
+                ModelState.AddModelError(string.Empty, "Unable to save changes.");
+            }
+            return View(book);
+        }
+
+        private Task<Book> FindBookAsync(int id)
+        {
+            return DbContext.Books.SingleOrDefaultAsync(x => x.BookID == id);
+        }
+
+        private IEnumerable<SelectListItem> GetAuthorsListItems(int selected = -1)
+        {
+            var tmp = DbContext.Authors.ToList();  // Workaround for https://github.com/aspnet/EntityFramework/issues/325
```

Comment:
Is this needed? The bug you  linked to is about traversing nav props in query but you aren't traversing any here.